src/Adapter/MSTestAdapter.PlatformServices/ObjectModel/UnitTestElement.cs

```diff
@@ -210,13 +211,9 @@ private Guid GenerateSerializedDataStrategyTestId(string testFullName)
 
         idProvider.AppendString(fileNameOrFilePath);
         idProvider.AppendString(testFullName);
-
         if (TestMethod.SerializedData != null)
         {
-            foreach (string? item in TestMethod.SerializedData)
-            {
-                idProvider.AppendString(item ?? "null");
-            }
+            idProvider.AppendString($"#{TestMethod.TestCaseIndex.ToString(CultureInfo.InvariantCulture)}#");
```

Comment:
Let's go with `[<index>]` syntax:

```suggestion
            idProvider.AppendString($"[{TestMethod.TestCaseIndex.ToString(CultureInfo.InvariantCulture)}]");
```